### DISCUSSION TITLE
Reports are not generating

### DISCUSSION THREAD

**Discussion by Venkatesh-Dayani23** [Sunbird-Obsrv/Community#126] [Issues]
Hi Team,
We are using Sunbird 5.0.0 version and we are unable to generate the reports we are getting the below error in the logs of analytics service. progress-exhaust is not available in dataset_sub_id in dev_dataset_metadata table.Kindly let us know how to fix this issue.

[error] s.StatementExecutor$$anon$1 - SQL execution failed (Reason: ERROR: relation "dev_dataset_metadata" does not exist           ││   Position: 15):                                                                                                                    ││                                                                                                                                     ││    select * from dev_dataset_metadata where dataset_sub_id = 'progress-exhaust'                                                     ││                                                                                                                                     ││ [error] a.a.OneForOneStrategy - ERROR: relation "dev_dataset_metadata" does not exist                                               ││   Position: 15                                                                                                                      ││ org.postgresql.util.PSQLException: ERROR: relation "dev_dataset_metadata" does not exist  

cc: @raghupathiguduri

**Comment by manjudr** [comment]
@Venkatesh-Dayani23  - Is `dev_dataset_metadata` table provisioned properly in the Postgres database? Please share the schema with us once? also, do we have any records in the table?

**Reply by Venkatesh-Dayani23** [reply]
HI @manjudr 
Below are the details of the tables.
![Screenshot from 2023-09-21 12-02-36](https://github.com/Sunbird-Obsrv/Community/assets/128453924/95d12e01-229a-439c-b49b-b107cfac67bf)


Below is the output file of the table _**dev_dataset_metadata**_
[output.csv](https://github.com/Sunbird-Obsrv/Community/files/12681153/output.csv)

cc: @raghupathiguduri

**Reply by manjudr** [reply]
@Venkatesh-Dayani23  - Did you try executing the `SELECT * FROM "dev_dataset_metadata" WHERE dataset_sub_id = 'progress-exhaust';` through CLI? are we getting the same exception though CLI as well? Do we have a `progress-exhaust` record in the table?

**Reply by Venkatesh-Dayani23** [reply]
Hi @manjudr 
We tried this Query SELECT * FROM "dev_dataset_metadata" WHERE dataset_sub_id = 'progress-exhaust'; below is the output of this query. we dont have a progress-exhaust record in the table.

![Screenshot from 2023-09-25 16-52-57](https://github.com/Sunbird-Obsrv/Community/assets/128453924/2ead9f33-0860-48d5-890e-8636f087ced5)


![Screenshot from 2023-09-25 16-53-37](https://github.com/Sunbird-Obsrv/Community/assets/128453924/dbcea78c-da2a-4008-89c0-c376cf6727cf)

**Reply by manjudr** [reply]
Which job are you trying to execute? are you trying to submit the job to run for the 'progress-exhaust' job?

**Reply by Venkatesh-Dayani23** [reply]
@manjudr we are trying to submit the all exhaust reports and execute them.

**Reply by manjudr** [reply]
This issue is unclear to us.  Could you please elaborate on the issues? Could you please let me know how you can reproduce this issue with the steps? are you getting this issue while running the job? or while accessing the database manually? what job are you trying to run? why it's throwing an exception to progress-exhaust job where there is not record found in the db relevant to progress exhaust? 
Please elaborate with steps

**Reply by Venkatesh-Dayani23** [reply]
Hi @manjudr 
When we try to generate the exhaust reports in the portal we are getting below mentioned error in the analytics service logs.

**SQL execution failed (Reason: ERROR: relation "dev_dataset_metadata" does not exist ││ Position: 15): ││ ││ select * from dev_dataset_metadata where dataset_sub_id = 'progress-exhaust' ││ ││ [error] a.a.OneForOneStrategy - ERROR: relation "dev_dataset_metadata" does not exist ││ Position: 15 ││ org.postgresql.util.PSQLException: ERROR: relation "dev_dataset_metadata" does not exist.**

So we checked in the Postgres database and noticed that **dataset_sub_id = 'progress-exhaust**' is not available in the **dev_dataset_metadata**.
Kindly need your help on this how we can add the **progress-exhaust dataset_sub_id** in the **dev_dataset_metadata** table.

cc: @raghupathiguduri

**Comment by Venkatesh-Dayani23** [comment]
after re provisioning the Spark server this fixed the issue.
